akka-actor-typed-tests/src/test/scala/akka/actor/typed/SupervisionSpec.scala

```diff
@@ -428,14 +468,122 @@ class SupervisionSpec extends TestKit("SupervisionSpec") with TypedAkkaSpecWithS
       probe.expectMsg(Started)
     }
 
-    "stop when exception from MutableBehavior constructor" in {
+    "restart when deferred factory throws" in new FailingDeferredTestSetup(
+      failCount = 1, strategy = SupervisorStrategy.restart) {
+
+      EventFilter[ConstructionFailedException](occurrences = 1).intercept {
+        spawn(behv)
+        probe.expectMsg(StartFailed)
+        probe.expectMsg(Started)
+      }
+    }
+
+    "fail instead of restart when deferred factory throws unhandled" in new FailingUnhandledTestSetup(
```

Comment:
does this mean that if one defines a general exception type, such as `Exception` it will restart infinitely if any such exception is thrown from constructor?

Reply:
Yeah, this is what was discussed a bit at one of our dailies, not sure it makes sense for `Restart`.

Reply:
ok, you haven't implemented that part yet?
I agree that we must be careful to avoid restart loops.

Reply:
Nope, haven't updated yet